setup.py

```diff
@@ -4,33 +4,37 @@
 import os
 
 
-class MissingPkgconfig:
+class Missing:
     """Raise an exception only when trying to convert it to a string.
 
-    This allows commands like ``setup.py clean`` to work even when pkgconfig
+    This allows commands like ``setup.py clean`` to work even when the package
     is not present, but makes it fail when trying to build the extension.
     """
+
+    def __init__(self, name):
+        self.name = name
+
     def __str__(self):
         raise RuntimeError(
-            'The pkgconfig module was not found. Try upgrading pip to the latest '
-            'version and using it to install.')
+            ('The {} module was not found. Try upgrading pip to the latest '
```

Comment:
Is there a definitive release from pip which supports pyproject.toml - we could then check the release and give a more informative error message. (e.g. Your version X.XX is too old, please upgrade to at least Y.YY / build tooling {} was not installed, please check for errors above)

Reply:
The feature first appeared in pip 10, but there have been various improvements and bug-fixes in later versions of pip so it's difficult to say without testing when it would first have worked, and it might even depend on exactly how the user is installing it (OS, virtualenv versus user dir versus system, etc).

Reply:
Fair enough - In that case maybe just mention upgrade pip and/or check your build environment - i.e. make it more explicit that the failure is that a required build tool is not getting auto-installed by pip as it should be.

Reply:
I've written a longer message now - not sure if it's clearer though.